projboard/models/user.py

```diff
@@ -28,3 +28,62 @@ def get_user_by_nickname(nickname):
         except ObjectDoesNotExist:
             return None
         return user
+
+    @staticmethod
+    def create_user(email, password, name, nickname):
+        """
+         Method to create a new user,
+         but also checks if user is already exists
+         [get user either by mail or by nickname]
+        :param email: user's email
+        :param password: user's password
+        :param name: user's name
+        :param nickname: user's nickname
+        :return: new user
+        """
+        user = User(email=email, password=password, name=name, nickname=nickname)
+        # user = User.objects.get(email=email)
+        user.save()
+        return user
+
+    @staticmethod
+    def delete_user_by_nickname(nickname):
+        """
+         Method to delete a user by his nickname
+        :param nickname: nickname to get the user
+        :return:  TRUE/FALSE if the user deleted/not
+        """
+        try:
+            user = User.get_user_by_nickname(nickname=nickname)
+            user.delete()
+        except User.DoesNotExist:
+            return False
+        return True
+
+    @staticmethod
+    def get_user_articles_by_likes(nickname):
+        """
+        Method get nickname and return User object if exist, otherwise None
+        :param nickname: the nickname to search
+        :return: User object
+        """
+        # NOT FINISHED YET
+        try:
+            user = User.objects.get(nickname=nickname)
+        except User.DoesNotExist:
+            return None
+        return user
+
+    @staticmethod
+    def get_user_articles_by_views(nickname):
+        """
+        Method get nickname and return User object if exist, otherwise None
+        :param nickname: the nickname to search
+        :return: User object
+        """
+        # NOT FINISHED YET
+        try:
+            user = User.objects.get(nickname=nickname)
+        except User.DoesNotExist:
+            return None
+        return user
```

Comment:
Do not leave unfinished code, either remove it or remark it

Reply:
@avi-biton Done